lib/ansible/plugins/connection/ssh.py

```diff
@@ -676,7 +676,15 @@ def _send_initial_data(self, fh, in_data):
             fh.write(to_bytes(in_data))
             fh.close()
         except (OSError, IOError):
-            raise AnsibleConnectionFailure('SSH Error: data could not be sent to remote host "%s". Make sure this host can be reached over ssh' % self.host)
+            # The ssh connection may have already terminated at this point, with a more useful error
+            # Only raise AnsibleConnectionFailure if the ssh process is still alive
+            time.sleep(0.001)
+            ssh_process.poll()
+            if getattr(ssh_process, 'returncode', None) is None:
+                raise AnsibleConnectionFailure(
+                    'SSH Error: data could not be sent to remote host "%s". Make sure this host can be reached '
+                    'over ssh' % self.host
+                )
```

Comment:
The exception will be ignored if it does not match the condition. Is this expected?

Reply:
This is expected.  This exception should only be raised when the ssh process is still alive.  If the ssh process is dead, the exception should not be raised, and instead, we will end up using the error from the ssh process.

Error before this PR:
```
vivo | UNREACHABLE! => {
    "changed": false,
    "msg": "SSH Error: data could not be sent to remote host \"vivo\". Make sure this host can be reached over ssh",
    "unreachable": true
}
```

Error with this PR (which is the correct error):
```
vivo | FAILED! => {
    "changed": false,
    "module_stderr": "/bin/sh: 1: /usr/bin/missing: not found\n",
    "module_stdout": "",
    "msg": "The module failed to execute correctly, you probably need to set the interpreter.\nSee stdout/stderr for the exact error",
    "rc": 127
}
```